Updated RPM spec to install Papyrus Compare

eposse opened the pull request:
Added installIU and uninstallIU clauses to RPM spec to install/uninstall Papyrus Compare

Signed-off-by: Ernesto Posse <eposse@zeligsoft.com>

J17359 commented:
@eposse - Adding these lines will attempt to install the feature group, but the install itself will fail because of the missing packages I mentioned in https://github.com/ZeligsoftDev/CX4CBDDS/issues/123. In order to test, build the RPM with the new spec file changes you've made and attempt to install the RPM. You should get an error  similar to the following:

> Cannot complete the install because one or more required items could not be found.
> Software being installed: Papyrus Compare Feature (Incubation) 0.7.0.202004292206 (org.eclipse.papyrus.compare.feature.feature.group 0.7.0.202004292206)
> Missing requirement: Papyrus Compare Feature (Incubation) 0.7.0.202004292206 (org.eclipse.papyrus.compare.feature.feature.group 0.7.0.202004292206) requires 'org.eclipse.equinox.p2.iu; org.eclipse.emf.compare.diagram.gmf.feature.group [3.0.0,4.0.0)' but it could not be found

J17359 commented:
I was able to successfully install the RPM now. Great work Ernesto!

eposse commented:
@J17359 I've updated it with comments from @elder4p who reviewed it. I've removed the -(un)installIU clauses for the EMF Compare features, while leaving the -(un)installIU for Papyrus Compare. The former are unnecessary, as we now include them in the p2 repo. I've tested it and it works for me. If you could test this latest commit, we can merge it.

Thanks.

J17359 commented:
> @J17359 If you could test this latest commit, we can merge it.

@eposse I'm building the RPM now. Will get back to you in an hour or so.

J17359 commented:
@eposse I have confirmed the new RPM installs correctly on my system. Merge away!

J17359 commented:
WAIT! Actually, the uninstall section is incorrect. You added the uninstall to the ACTD block in the if statement in the preinstall scriptlet. It should go under the axcioma block of the if statement.

I don't really know why there are both at this point...they have the same code inside them if I'm not mistaken.

J17359 commented on releng/com.zeligsoft.dds4ccm.update.axcioma/rpm_build/zeligsoftCX_axcioma.spec:
```diff
@@ -95,6 +95,7 @@ elif [[ ${atcdInstalledFeature} != "" ]]; then 		# atcd is already installed
    -uninstallIU org.eclipse.cdt.native.feature.group \
    -uninstallIU org.eclipse.cdt.platform.feature.group \
    -uninstallIU org.eclipse.cdt.sdk.feature.group \
+   -uninstallIU org.eclipse.papyrus.compare.feature.feature.group \
```
This should be up in the axcioma portion of the if/else block. Not sure we really need an if/else here at all if the code is the same between the two...that's a question for @elder4p

eposse commented:
I've added the missing -uninstall to that branch if the id/then. I left it in the atcd branch because the post action would install it in both cases.

eposse commented:
Done. @J17359 I'll leave it to you to close the issue.
